### PR TITLE
 [eslint-plugin-exhaustive-deps] Fix exhaustive deps check for unstable vars

### LINKED ISSUE #24268: Bug: [eslint-plugin-exhaustive-deps] can't find unstable value.
React version: 18.0.0 (not important)

## Steps To Reproduce
I'll show as a code.
스크린샷 2022-04-04 오전 12 01 05

Link to code example:
https://codesandbox.io/s/stupefied-raman-g6j3fi?file=/src/App.js

## The current behavior
exhaustive-deps can't found unstable dependency.

## The expected behavior
Warn lint message.

## Opinion
I know this code is super weird but This code may be improved. something like
- Add more core lint rule like 'useState is always declared as a const'
- eslint-plugin-exhaustive-deps should find reallocation value. and calculate real value.
- Or just stay it.

I just wonder what is your opinion. thank you.

### PR DESCRIPTION
**Pull request by afzalsayed96**
## Summary

Fixes #24268

## How did you test this change?

Added 2 test-cases, both valid and invalid.

### CODE REVIEW

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js [hunk 1]
```diff
@@ -234,6 +234,9 @@ export default {
             if (id.elements[1] === resolved.identifiers[0]) {
               if (name === 'useState') {
                 const references = resolved.references;
+                // console.log(references);
```
Comment (gaearon) [inline packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js:237]: Stray log

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js [hunk 2]
```diff
@@ -234,6 +234,9 @@ export default {
...
+                if (references.filter(ref => ref.isWrite()).length > 1)
```
Comment (gaearon) [inline packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js:238]: Please use curly braces for if statements

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js [hunk 3]
```diff
@@ -234,6 +234,9 @@ export default {
             if (id.elements[1] === resolved.identifiers[0]) {
               if (name === 'useState') {
                 const references = resolved.references;
+                if (references.filter(ref => ref.isWrite()).length > 1) {
```
[the remark this answers is not in the document]
Comment (afzalsayed96) [reply packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js]: Good catch

**Comment by gaearon** [comment]
Thanks!

**Pull request merged**